lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -453,6 +453,7 @@ def _validate_config(self, loader, path):
         if super(InventoryModule, self).verify_file(path):
             if path.endswith('.aws_ec2.yml' or '.aws_ec2.yaml'):
                 return self._read_config_data(path)
+            return {}
```

Comment:
when unable to parse a file you should raise an AnsibleParserError so the 'next plugin' can attempt processing.

Reply:
The difference that makes is if I specify a directory with a bad inventory file and a good one I get warnings. Should that be happening?